src/custom/pages/Swap/index.tsx

```diff
@@ -372,6 +375,85 @@ const SwapButton = ({ children, showLoading, showButton = false }: SwapButtonPro
     </Text>
   )
 
+interface HighFeeContainerProps {
+  padding?: string
+  margin?: string
+  width?: string
+}
+
+const HighFeeWarningContainer = styled(AuxInformationContainer).attrs((props) => ({
+  ...props,
+  hideInput: true,
+}))<HighFeeContainerProps>`
+  padding: ${({ padding = '5px 10px' }) => padding};
+  font-size: small;
+  font-weight: 400;
+  &&&&& {
+    background: #ff00008a;
+  }
+  width: ${({ width = '100%' }) => width};
+  border-radius: 5px;
+  margin: ${({ margin = '0 auto 12px auto' }) => margin};
+  > small {
+    display: flex;
+    justify-content: space-evenly;
+    align-items: center;
+    gap: 8px;
+    > div {
+      margin-top: 1px;
+    }
+
+    > span {
+      text-decoration: underline;
+      cursor: pointer;
+      margin-left: auto;
+    }
+  }
+  > div {
+    display: flex;
+    padding: 8px 0;
+  }
+`
+
+type HighFeeWarningProps = { trade?: TradeGp } & HighFeeContainerProps
+
+export const HighFeeWarning = (props: HighFeeWarningProps) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const infoLabel = useMemo(() => (isOpen ? '- info' : '+ info'), [isOpen])
+
+  // only considers inputAmount vs fee (fee is in input token)
+  const [isHighFee, feePercentage] = useMemo(() => {
+    if (!props.trade) return []
+
+    const feePercentage = props.trade.fee.feeAsCurrency.divide(props.trade.inputAmount).asFraction
+    return [feePercentage.greaterThan(FEE_SIZE_THRESHOLD), feePercentage.multiply('100')]
+  }, [props.trade])
+
+  if (!isHighFee) return null
+
+  return (
+    <HighFeeWarningContainer onClick={() => setIsOpen((state) => !state)} {...props}>
+      <small>
+        <AlertTriangle size={20} /> <div>Low yield swap detected!</div> <span>{infoLabel}</span>
```

Comment:
I'm not sold on `yield`.

To me, yield is something that you get as result of an investment, doesn't fit here.

What about: `Unfavorable trade` ?

Reply:
I think the component name gives good inspiration: "Warning: High fee detected!" :)

Reply:
Yeah this is contention

Reply:
So, are we keeping `yield`?

Reply:
It says unfavourable trade now